sources/test/test_container_storage_source.py

```diff
@@ -26,3 +26,14 @@ def test_containers_storage_integration(tmp_path, sources_module):
         sock = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
         cnt_storage = sources_module.ContainersStorageSource.from_args(["--service-fd", str(sock.fileno())])
         assert cnt_storage.exists(checksum, None)
+
+
+@pytest.mark.skipif(not has_executable("podman"), reason="no podman executable")
+@pytest.mark.skipif(os.getuid() != 0, reason="root only")
+def test_containers_storage_integration_missing(tmp_path, sources_module):
+    checksum = "sha256:1234567890123456789012345678901234567890909b14ffb032aa20fa23d9ad6"
+    sock = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
+    cnt_storage = sources_module.ContainersStorageSource.from_args(["--service-fd", str(sock.fileno())])
+    # this is not ideal, consider to just return "False" here
```

Comment:
This will also need it's own PR I feel - it would be nice to just return True/False here I think except that of course any unexpected skopeo errors should still be a real error. But right now it's all exit-code 1 so we cannot differenciate.  Some source code diving in skopeo is required I fear.

Reply:
I decided to actually address it here (in it's own commit of course) as it is simpler than expected. It relies on string parsing the error output from skopeo which is not great but it seems right now to be the only way, I did open https://github.com/containers/skopeo/pull/2236 to get some feedback from upstream if we could potentially get our own exit code.

Reply:
Last, last nitpick (sorry!). I missed this code comment comment because of the review comments - but I guess this comment isn't needed anymore inside the first function

Reply:
Thank you! No need to appologize, I love your attention to detail, really appreciated. This is fixed now.